app/src/main/java/com/example/dine_and_donate/UpcomingVouchersFragment.java

```diff
@@ -0,0 +1,85 @@
+package com.example.dine_and_donate;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+import com.example.dine_and_donate.Activities.HomeActivity;
+import com.example.dine_and_donate.Models.User;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class UpcomingVouchersFragment extends Fragment {
+
+    private RecyclerView recyclerView;
```

Comment:
mRecyclerView

Reply:
actually this prob doesn't need to be a field since you're only using it in onViewCreated. you can just declare it locally in that method